server/src/main/kotlin/com/tiden/flagship/circle/CircleClient.kt

```diff
@@ -60,8 +123,7 @@ suspend fun makePayment(paymentRequest: PaymentRequest): String {
     return responseBody
 }
 
-fun buildPaymentRequest(sourceId: String, sourceType: String, ipAddress: String, amount: String, verificationMethod: String, encryptedData: String, pubKeyId: String, description: String, email: String, phoneNumber: String, userSessionId: String): PaymentRequest {
-    val idempotencyKey = java.util.UUID.randomUUID().toString()
+fun buildPaymentRequest(idempotencyKey: String, sourceId: String, sourceType: String, ipAddress: String, amount: String, verificationMethod: String, encryptedData: String, pubKeyId: String, description: String, email: String, phoneNumber: String, userSessionId: String): PaymentRequest {
```

Comment:
passing in the `idempotencyKey` here -- it should come from the caller (likely router layer). For safely retrying API calls when things go wrong